manifests/mod/shib.pp

```diff
@@ -1,15 +1,30 @@
 class apache::mod::shib (
   $suppress_warning = false,
+  $use_external_repository = false,
 ) {
 
   if $::osfamily == 'RedHat' and ! $suppress_warning {
     warning('RedHat distributions do not have Apache mod_shib in their default package repositories.')
   }
 
+  if $::osfamily == 'Debian' and $use_external_repository {
+    if ! $suppress_warning {
+      warning('You have configure apache::mod::shib in a way that requires the configuration of')
+      warning('an external repository such as https://www.switch.ch/aai/guides/sp/installation')
+    }
+    $package = 'shibboleth'
+    $lib = 'mod_shib_22.so'
+  } else {
```

Comment:
Could you add an `elsif $use_external_repository {` that raises an error saying "use_external_repository is only supported on Debian osfamily" or something?